drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/factory/BrowserCapabilitiesList.java

```diff
@@ -230,5 +222,27 @@ public int getPrecedence() {
         }
     }
 
-    ;
+    public static class chromeHeadless implements BrowserCapabilities {
+        @Override
+        public String getReadableName() {
+            return "chromeHeadless";
+        }
+
+        @Override
+        public String getImplementationClassName() {
+            return "org.openqa.selenium.chrome.ChromeDriver";
+        }
+
+        @Override
+        public Map<String, ?> getRawCapabilities() {
+            DesiredCapabilities headlessChrome = DesiredCapabilities.chrome();
+            headlessChrome.setBrowserName("chromeHeadless");
```

Comment:
correct me if I'm wrong, this is `browserName` property, you check the `browser` property, which is different, so it shouldn't have any impact on your logic. 

>the browserName gets overridden somehow to `chrome`

browserName is by default set to `chrome` as you call `DesiredCapabilities.chrome()` where the property is set.